src/RobotInterface/src/Helper.cpp

```diff
@@ -499,6 +529,12 @@ bool RobotInterface::setPositionReferences(const iDynTree::VectorDynSize& desire
         return false;
     }
 
+    if(m_InteractionInterface == nullptr)
+        {
+            yError() << "[RobotInterface::setPositionReferences] IInteractionMode interface is not ready.";
+            return false;
+        }
```

Comment:
Since this check is used to avoid the segmentation-fault I would move the `if` statement before  `m_InteractionInterface->setInteractionModes(m_actuatedDOFs,jointsIndex.data(),modes.data());`